### PR TITLE
Adding a Makefile and bumping version of dependency

### PR DESCRIPTION
**Pull request by javaducky**
Very minor stuff while I was looking around. Figured I'd add a Makefile to make building locally a bit easier. 

Don't know enough on the CI setup just yet, so I didn't change the Go version in `go.mod`.

### CODE REVIEW

**Comment by MStoykov** [comment]
I really don't think the Makefile adds anything - most things in it replace `go <command>` with `make <command>` - so maybe add CONTRIBUTORS.md or INSTALLATION.md or update README.md :shrug: 

I also in general dislike Makefiles in go projects ... because in most cases they add nothing :rofl:

File: go.mod [hunk 1]
```diff
@@ -2,4 +2,4 @@ module go.k6.io/xk6
 
 go 1.16
 
-require github.com/Masterminds/semver/v3 v3.1.0
+require github.com/Masterminds/semver/v3 v3.1.1
```
[the remark this answers is not in the document]
Comment (javaducky) [reply go.mod:5]: No. Just a matter of being in there and wanting to confirm the addition of the `make vendor` was working correctly.

[the remark this answers is not in the document]
Comment (javaducky) [reply go.mod:5]: And I also saw that the xcaddy project had upgraded some time back.

**Review by imiric — APPROVED** [review]
Thanks Paul!

**Pull request merged**